WordPressFlux/WordPressFlux/Sources/QueryStore.swift

```diff
@@ -58,4 +113,62 @@ open class QueryStore<State, Query>: StatefulStore<State>, Unsubscribable {
     open func queriesChanged() {
         // Subclasses should implement this
     }
+
+}
+
+// In ideal world, those wouldn't be extensions on `Encodable`/`Codable`, but rather refinements of
+// `QueryStore` in which `Store` conforms to `Codable` — then we would just change behavior _inside_ the `QueryStore`,
+// based on whether the type conforms to `Codable` or not.
+// Regretfully, Swift type system (or my Swift skills) does't allow to express that —
+// you can't just do `let codableSelf = self as? QueryStore<Codable, Query>` because
+// _Codable itself doesn't conform to Codable_.
+private extension Encodable {
+    func storeCachedState(at url: URL) {
+        do {
+            let jsonEncoder = JSONEncoder.init()
+            let encodedStore = try jsonEncoder.encode(self)
+
+            try encodedStore.write(to: url, options: [.atomic])
+        } catch {
+            NSLog("[PluginStore Error] \(error)")
+        }
+    }
+}
+
+private extension Decodable {
+    static func cachedState(from cachedURL: URL) -> Self? {
+        do {
+            let data = try Data(contentsOf: cachedURL)
+            let state = try JSONDecoder().decode(Self.self, from: data)
```

Comment:
You can avoid the `do/catch` and use `try?/flatMap` here. I don't think we need to treat it as an error if the file doesn't exist, but we might want to log if it has invalid data.